src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -25858,6 +25858,176 @@ public void Test1(IDo<object> impl)
             Assert.Equal("IResult<System.Int32>", doSymbol.ReturnType.ToTestDisplayString());
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [WorkItem(40430, "https://github.com/dotnet/roslyn/issues/40430")]
+        public void MissingTypeArgumentInBase_ValueTuple(bool useImageReference)
+        {
+            var lib_cs = @"
+public class ClassWithTwoTypeParameters<T1, T2>
+{
+}
+
+public class SelfReferencingClassWithTuple
+    : ClassWithTwoTypeParameters<SelfReferencingClassWithTuple, (string A, int B)>
+{
+    public SelfReferencingClassWithTuple()
+    {
+        System.Console.Write(""ran"");
+    }
+}
+";
+            var lib = CreateCompilationWithMscorlib40(lib_cs, references: s_valueTupleRefs);
+            lib.VerifyDiagnostics();
+            var libRef = useImageReference ? lib.EmitToImageReference() : lib.ToMetadataReference();
+
+            var source_cs = @"
+public class TriggerStackOverflowException
```

Comment:
>TriggerStackOverflowException [](start = 13, length = 29)

Consider renaming this (as in later tests) since it shouldn't result in an exception.